src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/ResidentialChecksService.kt

```diff
@@ -148,20 +152,16 @@ class ResidentialChecksService(
     }
   }
 
-  private fun getAddressCheckStatus(addressCheckRequest: CurfewAddressCheckRequest): ResidentialChecksStatus {
-    val taskAnswers = addressCheckRequest.taskAnswers
-    val taskStatus = getPolicyTaskCodes().map { taskAnswers.find { answer -> answer.taskCode == it }.toTaskStatus() }
-
-    val unsuitable = taskStatus.any { it == UNSUITABLE }
-    val suitable = taskStatus.all { it == SUITABLE }
-    val inProgress = taskStatus.any { it == SUITABLE }
+  private fun isAddressSuitable(taskAnswers: MutableSet<ResidentialChecksTaskAnswer>): Boolean {
+    val taskStatus = getPolicyTaskCodes().map { taskAnswers.find { answer -> answer.taskCode == it }?.toTaskStatus() }
+    return taskStatus.all { it == SUITABLE }
```

Comment:
would change this to return a status rather than a boolean:

```
fun CurfewAddressCheckRequest.getStatus()
```
 then:
```kt
when {
    addressCheckRequests.any { it.status === SUITABLE } -> ResidentialChecksStatus.SUITABLE
    addressCheckRequests.all { it.status === UNSUITABLE } -> ResidentialChecksStatus.UNSUITABLE
    addressCheckRequests.any { it.status === IN_PROGRESS } -> ResidentialChecksStatus.IN_PROGRESS
   else -> ResidentialChecksStatus.NOT_STARTED


```